lib/puppet/network/http/connection.rb

```diff
@@ -197,6 +197,8 @@ def request_with_redirects(request, options)
     end
 
     def apply_options_to(request, options)
```

Comment:
We do not explicitly set a User-Agent in Puppet Server.  our client gets created here:

https://github.com/puppetlabs/puppetserver/blob/fa34cedb5ba38f86ab156b40c007fe25fae600b1/src/ruby/puppetserver-lib/puppet/server/http_client.rb#L162

And that ends up (through a couple of layers) instantiating an instance of the Apache async-http-client.  I don't believe that the Apache client provides a default value for that header so my guess is that we're not sending one at all.

If necessary we could certainly file a separate ticket for Puppet Server to have it add the header.